### ISSUE TITLE
Missing true_pos for label: 1

### ISSUE THREAD

**Issue by fromwhzz** [weiliu89/caffe#46]
Hi:
  I train new models from new data. the new data has four label(including background 0). but I find it will not detect label 1 and label 2 object, it will only detect label 3 object. And I check the training log. I find in the training log , all the test net will print 
I0715 09:58:41.775827   257 solver.cpp:421] Iteration 1000, Testing net (#0)
I0715 09:58:42.032855   257 net.cpp:684] Ignoring source layer mbox_loss
W0715 09:58:44.846182   257 solver.cpp:507] Missing true_pos for label: 1
W0715 09:58:44.846276   257 solver.cpp:507] Missing true_pos for label: 2
I0715 09:58:44.846349   257 solver.cpp:526]     Test net output #0: detection_eval = 0.333333

the last layer of loss is 
layer {
  name: "mbox_loss"
  type: "MultiBoxLoss"
  bottom: "mbox_loc"
  bottom: "mbox_conf"
  bottom: "mbox_priorbox"
  bottom: "label"
  top: "mbox_loss"
  include {
    phase: TRAIN
  }
  propagate_down: true
  propagate_down: true
  propagate_down: false
  propagate_down: false
  loss_param {
    normalization: VALID
  }
  multibox_loss_param {
    loc_loss_type: SMOOTH_L1
    conf_loss_type: SOFTMAX
    loc_weight: 1.0
    num_classes: 4
    share_location: true
    match_type: PER_PREDICTION
    overlap_threshold: 0.5
    use_prior_for_matching: true
    background_label_id: 0
    use_difficult_gt: true
    do_neg_mining: true
    neg_pos_ratio: 3.0
    neg_overlap: 0.5
    code_type: CENTER_SIZE
  }

And I check the lmdb data, I find the label is right , in the training and testing data, it has label 1 and label 2 object.
Is there any problem in training?
thanks.

**Comment by fromwhzz** [comment]
the problem is because the data is wrong, thanks.

**Comment by leejiajun** [comment]
@weiliu89 
@fromwhzz 

I changed network, but I got this message. 
why?

> W0911 00:43:53.443533 20747 solver.cpp:512] Missing true_pos for label: 1
W0911 00:43:53.443758 20747 solver.cpp:512] Missing true_pos for label: 2
W0911 00:43:53.443768 20747 solver.cpp:512] Missing true_pos for label: 3
W0911 00:43:53.443771 20747 solver.cpp:512] Missing true_pos for label: 4
W0911 00:43:53.443778 20747 solver.cpp:512] Missing true_pos for label: 5
W0911 00:43:53.443781 20747 solver.cpp:512] Missing true_pos for label: 6
W0911 00:43:53.514109 20747 solver.cpp:512] Missing true_pos for label: 8
W0911 00:43:53.515017 20747 solver.cpp:512] Missing true_pos for label: 10
W0911 00:43:53.515419 20747 solver.cpp:512] Missing true_pos for label: 12
W0911 00:43:53.515429 20747 solver.cpp:512] Missing true_pos for label: 13
W0911 00:43:53.515434 20747 solver.cpp:512] Missing true_pos for label: 14
W0911 00:43:53.529917 20747 solver.cpp:512] Missing true_pos for label: 16
W0911 00:43:53.529930 20747 solver.cpp:512] Missing true_pos for label: 17
W0911 00:43:53.529935 20747 solver.cpp:512] Missing true_pos for label: 18
W0911 00:43:53.529940 20747 solver.cpp:512] Missing true_pos for label: 19
W0911 00:43:53.529944 20747 solver.cpp:512] Missing true_pos for label: 20

**Comment by leejiajun** [comment]
@fromwhzz 
what is wrong of your data?

**Comment by ChristineRYY** [comment]
@leejiajun  I use the alex based ssd model in your CSDN blog to tain voc0712 dataset. I also got this output, and the detection_eval is lower than 0.01. Can you tell me how to solve it? Thanks.

**Comment by leejiajun** [comment]
@ChristineRYY could you give me more detail?

**Comment by ChristineRYY** [comment]
@leejiajun Thanks. I finetune it on existing caffemodel, and the problem is solved.

**Comment by abhisheksgumadi** [comment]
I investigated into this and I think this message is thrown when the test data does not contain samples of the classes referenced in the dataset. I believe this has nothing to do with the training process.

**Comment by jtr2012** [comment]
@fromwhzz I have the same problem, have you fixed it?
